File: Linked List /Python/3. 160. Intersection of Two Linked Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        tempA, tempB = headA, headB
        while(tempA != tempB):
            tempA = tempA.next if tempA is not None else headB
            tempB = tempB.next if tempB is not None else headA
        return tempA

File: Linked List /C++/3. 160. Intersection of Two Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode *tempA = headA;
        ListNode *tempB = headB;
        while(tempA != tempB)
        {
            if(tempA != NULL)
            {
                tempA = tempA -> next;
            }
            else
            {
                tempA = headB;
            }

            if(tempB != NULL)
            {
                tempB = tempB -> next;
            }
            else
            {
                tempB = headA;
            }
        }
        return tempA;
    }
};
